src/libraries/System.Private.CoreLib/src/System/Environment.cs

```diff
@@ -210,6 +212,19 @@ public static string StackTrace
             get => new StackTrace(true).ToString(System.Diagnostics.StackTrace.TraceFormat.Normal);
         }
 
+        public static int SystemPageSize
+        {
+            get
+            {
+                int systemPageSize = s_systemPageSize;
+                if (systemPageSize == default)
+                {
+                    Interlocked.CompareExchange(ref s_systemPageSize, systemPageSize = GetSystemPageSize(), default);
```

Comment:
Is Interlocked really needed here? I guess nothing bad will happen if we call GetSystemPageSize twice
It will make `get_SystemPageSize` smaller and more inline-friendly

Reply:
> Is Interlocked really needed here?

Not really, but I wanted to be consistent with other cached `Environment` properties:

https://github.com/dotnet/runtime/blob/59b6c36b35553a63e4bf1178922ff97d90540220/src/libraries/System.Private.CoreLib/src/System/Environment.cs#L128-L131

> It will make get_SystemPageSize smaller and more inline-friendly

Yes, but on the other hand inlining this method compared to getting rid of the sys-call is a very minor perf improvement.

Reply:
It's not necessary unless there's a possibility it might change,  which I don't believe there is.  Here and with process ID, better to get rid of the interlocked and just make the field volatile.

Reply:
Also,  the way it's currently written with the assignment to the local as part of the interlocked defeats any protection the interlocked was there to achieve around ensuring idempotency in the face of an underlying value changing.